applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/hcf_data_container.h

```diff
@@ -0,0 +1,204 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alireza Taherzadeh-Fard
+//                   Alejandro Cornejo
+//
+//
+
+# pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/constitutive_law.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class HCFDataContainer
+ * @ingroup ConstitutiveLawsApplication
+ * @brief Defining all the methods and variables required in fatigue simulations
+ * @details Give access to methods and restores all member variables in fatigue simulations
+ * @author Alireza Taherzadeh-Fard, Alejandro Cornejo
+ */
+class HCFDataContainer
+{
+public:
+
+    struct FatigueVariables {
+        double MaxStress = 0.0;
+        double MinStress = 0.0;
+        bool MaxIndicator = false;
+        bool MinIndicator = false;
+        Vector PreviousStresses = ZeroVector(2);
+        double FatigueReductionFactor = 1.0;
+        double ReversionFactorRelativeError = 0.0;
+        double MaxStressRelativeError = 0.0;
+        unsigned int GlobalNumberOfCycles = 1;
+        unsigned int LocalNumberOfCycles = 1;
+        double B0 = 0.0;
+        double PreviousMaxStress = 0.0;
+        double PreviousMinStress = 0.0;
+        double WohlerStress = 1.0;
+        double Sth = 0.0;
+        double CyclesToFailure = 0.0;
+        bool NewCycle = false;
+        double Alphat = 0.0;
+        double PreviousReversionFactor = 0.0;
+        double ReversionFactor = 0.0;
+        bool AdvanceStrategyApplied;
+        bool DamageActivation;
+    };
+
+    static constexpr double tolerance = 1.0e-3;
+
+    HCFDataContainer()
+    {};
+
+    // Defining fatigue methods
+
```

Comment:
BTW I think that we need a new method called `Check` which ensures that the material properties have the required Fatigue parameters in them.

Reply:
to be called in the `Check` from the CL